### PR TITLE
express-did-auth - Implement personal_sign user signature

### PR DESCRIPTION
**Pull request by ilanolkies**
This PR enforces the user to sign using `personal_sign` instead of JWTs. Reference: https://github.com/rsksmart/rif-identity-docs/pull/28

Status of the implementation:
- [x] Implement a happy path using `personal_sign`
- [x] Remove skipped tests
- [x] Correct tests to use new signature kind
- [x] Validate all error messages are still correct
- [x] Clean unused code - removed code will keep commented until this phase

### CODE REVIEW

File: packages/express-did-auth/src/factories/authentication-factory.ts [hunk 1]
```diff
@@ -23,27 +21,34 @@ export function authenticationFactory (
...
 
-      const isValid = businessLogic ? await businessLogic(payload) : true
+      const address = '0x' + pubToAddress(ecrecover(
+        messageDigest,
+        ecdsaSignature.v,
+        ecdsaSignature.r,
+        ecdsaSignature.s
+      )).toString('hex')
+
+      if (address !== did.split(':').pop().toLowerCase()) return res.status(401).send(INVALID_CHALLENGE_RESPONSE)
```
[the remark this answers is not in the document]
Comment (ilanolkies) [reply packages/express-did-auth/src/factories/authentication-factory.ts:38]: It should, no?

File: packages/express-did-auth/test/authentication-factory.test.ts [hunk 2]
```diff
@@ -51,71 +54,100 @@ describe('authenticationFactory', () => {
     await testAuthFactory(state)(req, res)
   })
 
-  test('should respond with 401 if the subject of the challenge response is not the service did', async () => {
+  test('should respond with 401 if the subject of the challenge response is not the service url', async () => {
```
Comment (javiesses) [inline packages/express-did-auth/test/authentication-factory.test.ts:57]:
```suggestion
  test('should respond with 401 if the signed message contains another service url', async () => {
```

File: packages/express-did-auth/test/authentication-factory.test.ts [hunk 3]
```diff
@@ -51,71 +54,100 @@ describe('authenticationFactory', () => {
...
     const req = { body: { response: challengeResponseJwt } }
-    const res = mockedResFactory(401, CORRUPTED_CHALLENGE)
+    const res = mockedResFactory(401, INVALID_CHALLENGE_RESPONSE)
 
     const logic = mockBusinessLogicFactory(false)
     await testAuthFactory(state, logic)(req, res)
   })
 
-  test('should respond with 401 if extra business logic that returns false', async () => {
+  test('should respond with 401 if the subject of the signer is not the given did', async () => {
```
Comment (javiesses) [inline packages/express-did-auth/test/authentication-factory.test.ts:71]:
```suggestion
  test('should respond with 401 if the message is signed with another did', async () => {
```

File: packages/express-did-auth/test/authentication-factory.test.ts [hunk 4]
```diff
@@ -51,71 +54,100 @@ describe('authenticationFactory', () => {
...
     MockDate.set(modulo0Timestamp)
 
     const challenge = challengeVerifier.get(userIdentity.did)
-    const challengeResponseJwt = await challengeResponseFactory(challenge, userIdentity, config.serviceDid, config.serviceUrl)
+    const anotherIdentity = await identityFactory()
+
+    const challengeResponseJwt = challengeResponseFactory(challenge, anotherIdentity.identity, anotherIdentity.privateKey, 'https://taringa.net')
```
Comment (javiesses) [inline packages/express-did-auth/test/authentication-factory.test.ts:77]:
```suggestion
    const challengeResponseJwt = challengeResponseFactory(challenge, anotherIdentity.identity, anotherIdentity.privateKey, config.serviceUrl)
```
Let's use the proper serviceUrl so we can test that is validating just the signer

File: packages/express-did-auth/test/authentication-factory.test.ts [hunk 5]
```diff
@@ -51,71 +54,100 @@ describe('authenticationFactory', () => {
...
     const req = { body: { response: challengeResponseJwt } }
-    const res = mockedResFactory(401, UNAUTHORIZED_USER)
+    const res = mockedResFactory(401, INVALID_CHALLENGE_RESPONSE)
 
     const logic = mockBusinessLogicFactory(false)
     await testAuthFactory(state, logic)(req, res)
   })
 
-  test('should respond with 401 if extra business logic that throws an error', async () => {
+  test('should respond with 401 if the subject of the signer is not the specified did', async () => {
```
Comment (javiesses) [inline packages/express-did-auth/test/authentication-factory.test.ts:86]:
```suggestion
  test('should respond with 401 if the signer of the message is not the specified did', async () => {
```

File: packages/express-did-auth/test/authentication-factory.test.ts [hunk 6]
```diff
@@ -51,71 +54,100 @@ describe('authenticationFactory', () => {
...
     let req, res
 
     beforeEach(async () => {
       MockDate.set(modulo0Timestamp)
 
       const challenge = challengeVerifier.get(userIdentity.did)
-      const challengeResponseJwt = await challengeResponseFactory(challenge, userIdentity, config.serviceDid, config.serviceUrl)
-
-      req = { body: { response: challengeResponseJwt } }
+      const response = await challengeResponseFactory(challenge, userIdentity, userPrivateKey, config.serviceUrl)
```
Comment (javiesses) [inline packages/express-did-auth/test/authentication-factory.test.ts:149]:
```suggestion
      const response = challengeResponseFactory(challenge, userIdentity, userPrivateKey, config.serviceUrl)
```

File: packages/express-did-auth/test/setup-app.cookies.integration.test.ts [hunk 7]
```diff
@@ -38,7 +42,7 @@ describe.skip('Express app tests (using cookies)', () => {
   })
 
   it('2. POST /signup', async () => {
-    const challengeResponse = await challengeResponseFactory(challenge, userIdentity, serviceDid, serviceUrl)
+    const challengeResponse = await challengeResponseFactory(challenge, userIdentity, userPrivateKey, serviceUrl)
```
Comment (javiesses) [inline packages/express-did-auth/test/setup-app.cookies.integration.test.ts:45]:
```suggestion
    const challengeResponse = challengeResponseFactory(challenge, userIdentity, userPrivateKey, serviceUrl)
```

File: packages/express-did-auth/test/setup-app.cookies.integration.test.ts [hunk 8]
```diff
@@ -55,7 +59,7 @@ describe.skip('Express app tests (using cookies)', () => {
   })
 
   it('4. POST /auth', async () => {
-    const challengeResponse = await challengeResponseFactory(challenge, userIdentity, serviceDid, serviceUrl)
+    const challengeResponse = await challengeResponseFactory(challenge, userIdentity, userPrivateKey, serviceUrl)
```
Comment (javiesses) [inline packages/express-did-auth/test/setup-app.cookies.integration.test.ts:62]:
```suggestion
    const challengeResponse = challengeResponseFactory(challenge, userIdentity, userPrivateKey, serviceUrl)
```

**Review by javiesses — COMMENTED** [review]
Like it! Minor changes requested

File: packages/express-did-auth/src/factories/authentication-factory.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ilanolkies) [reply packages/express-did-auth/src/factories/authentication-factory.ts:38]: Let's keep it this way now. Once we start integrating we will find the correct expected input. What do you think?

[the remark this answers is not in the document]
Comment (javiesses) [reply packages/express-did-auth/src/factories/authentication-factory.ts:38]: Agree

File: packages/express-did-auth/test/authentication-factory.test.ts (hunk 4 above)
Comment (ilanolkies) [reply packages/express-did-auth/test/authentication-factory.test.ts:77]: Uh sorry. Copy paste 🤦

**Review by javiesses — APPROVED** [review]
LGTM

**Pull request merged**